consensus/src/detail/spend_cache.rs

```diff
@@ -165,13 +176,86 @@ impl<'a> CachedInputs<'a> {
         Ok(())
     }
 
+    fn verify_signatures(&self, tx: &Transaction) -> Result<(), BlockError> {
+        for (input_idx, input) in tx.get_inputs().iter().enumerate() {
+            let outpoint = input.get_outpoint();
+            let prev_tx_index_op = self.get_from_cached(outpoint)?;
+
+            let tx_index = prev_tx_index_op
+                .get_tx_index()
+                .ok_or(BlockError::PreviouslyCachedInputNotFound)?;
```

Comment:
Is this a verification error or a code invarinat? If the latter, maybe an `expect` would be more appropriate?

Reply:
I'm torn in these because I still want the final form to be available before making such a commitment. I'm always afraid we'd mess this up and change some order of things and then things break and crash the node on demand by an attacker.